Automattic/woocommerce-payments
# Avoid product service exceptions during checkout

Fixes #6645

#### Changes proposed in this Pull Request

Make sure that the product service doesn't assume everything during the `save_post` action is a product by using the specific `save_post_product` action. This way during checkout, while creating an order, there are no exceptions.

#### Testing instructions

1. Listen to Xdebug, and all exceptions.
2. Without this branch, test checkout. You'll see exceptions.
3. Check out this branch.
4. Try checkout again, there should be no exceptions.

No tests here, there is nothing new in the PR.

-------------------
